src/utilities/guards.ts

```diff
@@ -19,3 +19,7 @@ export function hasProperty<
 
   return true
 }
+
+export function isNestedArray<T extends any[] | readonly any[]>(value: T | T[]): value is T[] {
+  return value.length > 0 && Array.isArray(value[0])
```

Comment:
Should this maybe be `value.every(item => Array.isArray(item))`

Reply:
I suppose that's perhaps technically more accurate given that this is a utility, but in the context of createRouter, if the 1st item in the array is an array, they are all going to be. I'd prefer not to iterate through potentially a large set to prove what Typescript types should have already enforced?

Reply:
I don't feel strongly. But it's probably a small array and it reads better IMO. Merge away as is 👍